### ISSUE TITLE
sd_service: Fails to restart if service is failed

### ISSUE THREAD

**Issue by jilleJr** [jetporch/jetporch#10]
### What Happened?

Running `!sd_service` with `restart: true` fails if the service is failed.

The service can be failed for many reasons, but most commonly the application its running is just exiting with non-zero exit code.

The error is originating from this piece of code:

https://github.com/jetporch/jetporch/blob/6a78802d5cc961094edb90c1c803b0d27d34bd8e/src/modules/services/sd_service.rs#L148-L154

Running `systemctl is-active` on a failed unit returns the string `failed`

### Version

```
┌─────┬─────────────────────────────────────────────┐
│jetp │http://www.jetporch.com/                     │
│     │(C) Michael DeHaan + contributors, 2023      │
│     │                                             │
│build│6a78802d5cc961094edb90c1c803b0d27d34bd8e@main│
│     │Sat Sep 30 05:49:44 PM CEST 2023             │
├─────┼─────────────────────────────────────────────┤
│     │usage: jetp <MODE> [flags]                   │
└─────┴─────────────────────────────────────────────┘
```

### Steps to Reproduce

1. Have a host with a failing systemd unit

2. Have a playbook with the following:

   ```yaml
   - name: systemd playbook
     groups: [vms]
     sudo: root
     tasks:
       - !sd_service
         service: jetp-failing-test
         restart: true
   ```

3. Run playbook, and get the following error:

   ```console
   $ jetp ssh -i inventory -p playbooks/systemd-unit-restart.yml
   ----------------------------------------------------------
   > playbook start: /home/kalle/code/jetp-testing/playbooks/systemd-unit-restart.yml
   ----------------------------------------------------------
   > play: systemd playbook
   ----------------------------------------------------------
   > batch 1/1, 1 hosts
   ----------------------------------------------------------
   > begin task: sd_service
   … 192.168.8.120 => running
   ! error: 192.168.8.120: systemctl activity status unexpected for service(jetp-failing-test): failed
   ----------------------------------------------------------
   > play complete: systemd playbook
   ----------------------------------------------------------
   
   ┌─────────┬─────┬─────┐
   │Results  │Items│Hosts│
   ├─────────┼─────┼─────┤
   │Roles    │0    │     │
   │Tasks    │1    │1    │
   ├─────────┼─────┼─────┤
   │Matched  │0    │0    │
   │Created  │0    │0    │
   │Modified │0    │0    │
   │Removed  │0    │0    │
   │Executed │0    │0    │
   │Passive  │0    │0    │
   │Skipped  │0    │0    │
   ├─────────┼─────┼─────┤
   │Unchanged│0    │1    │
   │Changed  │0    │0    │
   │Failed   │1    │1    │
   └─────────┴─────┴─────┘
   
   (X) Failures have occured.
   ```

**Comment by mpdehaan** [comment]
Yep, I definitely didn't have a failed service when I was testing this one, thanks for catching... nicely formatted output is much appreciated.

This is an easy fix so will take care of it soon if no one else beats me to it.  Thank you!

**Comment by mpdehaan** [comment]
Fix push in 5409b8b5a0858ef23197fd132be46d867ad6deac ... thanks again for the report!
